src/ILCompiler.WebAssembly/src/Compiler/DependencyAnalysis/WebAssemblyMethodCodeNode.cs

```diff
@@ -59,8 +59,17 @@ public override IEnumerable<DependencyListEntry> GetStaticDependencies(NodeFacto
         {
             var dependencies = new DependencyList();
 
-            foreach (Object node in _dependencies)
-                dependencies.Add(node, "Wasm code ");
+            if (_method.HasInstantiation && !factory.CanNecessaryTypeSymbol(_method.OwningType))
```

Comment:
I'm a bit worried about this line.

Is it possible the need for this if block comes from the metadata policy needing tweaks? Do we still need it after addressing the other feedback?

Reply:
With the policy change, and this reverted it hits the assert at https://github.com/dotnet/corert/blob/74897d4ab3eaff1141f8ac07287ccefbfbb5a352/src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ExactMethodInstantiationsNode.cs#L85 with stack
```
I'll do some more investigation on this one as this came up in the dev and there were some comments here https://github.com/dotnet/corert/issues/5842#issuecomment-430605137 and in your reply which I'll go over.
         Assertion Failed

            at ILCompiler.DependencyAnalysis.ExactMethodInstantiationsNode.GetData(NodeFactory factory, Boolean relocsOnly) in D:\GitHub\corert\src\ILCompiler.Compiler\src\Compiler\DependencyAnalysis\ExactMethodInstantiationsNode.cs:line 85
            at ILCompiler.DependencyAnalysis.WebAssemblyObjectWriter.EmitObject(String objectFilePath, IEnumerable`1 nodes, NodeFactory factory, WebAssemblyCodegenCompilation compilation, IObjectDumper dumper) in D:\GitHub\corert\src\ILCompiler.WebAssembly\src\CodeGen\WebAssemblyObjectWriter.cs:line 772
            at ILCompiler.WebAssemblyCodegenCompilation.CompileInternal(String outputFile, ObjectDumper dumper) in D:\GitHub\corert\src\ILCompiler.WebAssembly\src\Compiler\WebAssemblyCodegenCompilation.cs:line 54
            at ILCompiler.Compilation.ILCompiler.ICompilation.Compile(String outputFile, ObjectDumper dumper) in D:\GitHub\corert\src\ILCompiler.Compiler\src\Compiler\Compilation.cs:line 387
            at ILCompiler.Program.Run(String[] args) in D:\GitHub\corert\src\ILCompiler\src\Program.cs:line 576
            at ILCompiler.Program.Main(String[] args) in D:\GitHub\corert\src\ILCompiler\src\Program.cs:line 732
```

Reply:
I'll do some more investigation here as this came up in the dev here https://github.com/dotnet/corert/issues/5842#issuecomment-430605137 and in your response.

Reply:
Its not quite the same, this time the method with no `SavedVertex` is `{[S.P.CoreLib]Internal.Runtime.CompilerServices.Unsafe.As<uint8,object>(uint8&)}` with the generic definition `{[S.P.CoreLib]Internal.Runtime.CompilerServices.Unsafe.As<TFrom,TTo>(!!0&)}`

Reply:
Let's make this condition into `if (factory.TypeSystemContext.IsUnboxingThunk(_method))`. The `IsUnboxingThunk` method doesn't exist yet, but there's `IsSpecialUnboxingThunk`.

The implementation will be the same except of the typeof check (instead of `GenericUnboxingThunk`, use `UnboxingThunk`).

Reply:
Or maybe wait a bit. I'll build a copy of what you have and look at all the asserts under a debugger. It will be more efficient that way.